packages/react-reconciler/src/ReactHookEffectTags.js

```diff
@@ -9,11 +9,12 @@
 
 export type HookEffectTag = number;
 
-export const NoEffect = /*             */ 0b00000000;
-export const UnmountSnapshot = /*      */ 0b00000010;
-export const UnmountMutation = /*      */ 0b00000100;
-export const MountMutation = /*        */ 0b00001000;
-export const UnmountLayout = /*        */ 0b00010000;
-export const MountLayout = /*          */ 0b00100000;
-export const MountPassive = /*         */ 0b01000000;
-export const UnmountPassive = /*       */ 0b10000000;
+export const NoEffect = /*                   */ 0b000000000;
+export const UnmountSnapshot = /*            */ 0b000000010;
+export const UnmountMutation = /*            */ 0b000000100;
+export const MountMutation = /*              */ 0b000001000;
+export const UnmountLayout = /*              */ 0b000010000;
+export const MountLayout = /*                */ 0b000100000;
+export const MountPassive = /*               */ 0b001000000;
+export const UnmountPassive = /*             */ 0b010000000;
+export const NoEffectPassiveUnmountFiber = /**/ 0b100000000;
```

Comment:
The name of the new effect tag I added, `NoEffectPassiveUnmountFiber`, is kind of stupid. I'd welcome a solution for a better name. 🤪

Reply:
I think it should be possible to simplify these tags by removing the Mount/Unmount . I think that's already handled because of `commitHookEffectList` accepts two separate arguments for which tags to mount and which tags to unmount.

Would it work if the only tags we had were this?

```js
// Represents whether effect should fire.
export const HasEffect = /*  */ 0b0001;
// Represents the phase in which the effect (not the clean-up) fires.
export const Snapshot = /*   */ 0b0010;
export const Layout = /*     */ 0b0100;
export const Passive = /*    */ 0b1000;
```

So during an update you would fire the clean up of any hook that has (for passive effects) that has `Passive | HasEffect`. When the fiber is deleted, you would only check `Passive`.

Reply:
I think this cleanup suggestion makes sense, although while looking into it I realized we have a potential bug with the sequence/timing of passive effects. (We don't call all destroy functions before create functions.) I'll take a pass at the effect tag cleanup *and* separating the passive effect destroy/create phases too. If it starts looking like that might be a bigger change though, I may split it out into a follow up PR.